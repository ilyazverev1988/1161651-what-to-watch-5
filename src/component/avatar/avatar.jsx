import React, {Fragment} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ButtonShowMore} from "../button-show-more/button-show-more";

const Avatar = (props) => {
  const {userData, authorizationStatus} = props;
  const avatarUrl = userData.avatar_url;
  return (
    <Fragment>
      {authorizationStatus === `AUTH` ?
        <Fragment>
          <div className="user-block">
            <div className="user-block__avatar">
              <img src={avatarUrl} alt="User avatar" width="63" height="63"/>
            </div>
          </div>
        </Fragment> : ``}
    </Fragment>
  );
};

const mapStateToProps = ({USER}) => ({
  userData: USER.userData,
  authorizationStatus: USER.authorizationStatus
});

export {Avatar};
export default connect(mapStateToProps, null)(Avatar);
