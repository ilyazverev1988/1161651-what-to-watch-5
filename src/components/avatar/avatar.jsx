import React, {Fragment} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

const Avatar = (props) => {
  const {userData, authorizationStatus} = props;
  const avatarUrl = userData.avatar_url;
  return (
    <Fragment>
      {authorizationStatus === `AUTH` ?
        <Fragment>
          <div className="user-block">
            <div className="user-block__avatar">
              <Link to={`/mylist`}>
                <img src={avatarUrl} alt="User avatar" width="63" height="63"/>
              </Link>
            </div>
          </div>
        </Fragment> :
        <Fragment>
          <div className="user-block">
            <Link to={`/login`} href="sign-in.html" className="user-block__link">Sign in</Link>
          </div>
        </Fragment>}
    </Fragment>
  );
};

Avatar.propTypes = {
  userData: PropTypes.oneOfType([PropTypes.string, PropTypes.shape(undefined)]),
  authorizationStatus: PropTypes.string.isRequired,
};

const mapStateToProps = ({USER}) => ({
  userData: USER.userData,
  authorizationStatus: USER.authorizationStatus
});

export {Avatar};
export default connect(mapStateToProps, null)(Avatar);
