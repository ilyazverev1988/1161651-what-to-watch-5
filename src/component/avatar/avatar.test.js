import React from "react";
import renderer from "react-test-renderer";
import {Avatar} from "./avatar";
import {MemoryRouter} from "react-router-dom";

const userData = {
  id: 1,
  email: `3@mail.ru`,
  name: `3`,
  avatarURL: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
  errorAuthorization: ``
};


describe(`Should AuthScreen render correctly`, () => {
  it(`With Autorization`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <Avatar
              userData={userData}
              authorizationStatus={`AUTH`}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Without Autorization`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <Avatar
              userData={``}
              authorizationStatus={`NO AUTH`}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
