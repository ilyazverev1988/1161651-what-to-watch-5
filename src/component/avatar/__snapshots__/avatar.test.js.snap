// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AuthScreen render correctly With Autorization 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        width="63"
      />
    </a>
  </div>
</div>
`;

exports[`Should AuthScreen render correctly Without Autorization 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
